weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx

```diff
@@ -0,0 +1,1936 @@
+  withFileTag,
+  withoutTags,
+} from '@wandb/weave/core';
+import {produce} from 'immer';
+import _ from 'lodash';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import {View as VegaView, VisualizationSpec} from 'react-vega';
+import {calculatePosition} from 'vega-tooltip';
+
+import {
+  useWeaveContext,
+  useWeaveRedesignedPlotConfigEnabled,
+} from '../../../context';
+import * as LLReact from '../../../react';
+import {getPanelStackDims, getPanelStacksForType} from '../availablePanels';
+import {Panel2Loader, PanelComp2} from '../PanelComp';
+import {usePanelContext} from '../PanelContext';
+import {makeEventRecorder} from '../panellib/libanalytics';
+import * as TableState from '../PanelTable/tableState';
+import {toWeaveType} from '../toWeaveType';
+import {ensureValidSignals, useConcreteConfig, useConfig} from './config';
+import {filterTableNodeToSelection} from './filter';
+import * as PlotState from './plotState';
+import {Portal} from './Portal';
+import {PanelPlotRadioButtons} from './RadioButtons';
+import {BrushMode, PanelPlotProps} from './types';
+import {defaultPlot, getAxisTimeUnit} from './util';
+import {
+  getMark,
+  stringHashCode,
+  stringIsColorLike,
+  useVegaReadyTables,
+} from './util';
+import {
+  ConcretePlotConfig,
+  LINE_SHAPES,
+  MarkOption,
+  PlotConfig,
+  POINT_SHAPES,
+  Scale,
+  SeriesConfig,
+} from './versions';
+
+const recordEvent = makeEventRecorder('Plot');
+
+/* eslint-disable no-template-curly-in-string */
+const PLOT_TEMPLATE: VisualizationSpec = {
+  $schema: 'https://vega.github.io/schema/vega-lite/v4.json',
+  data: {
+    name: 'wandb',
+  },
+  padding: 1,
+  title: '${field:title}',
+  mark: {
+    // tooltip: true,
+  } as any,
+  params: [],
+  encoding: {
+    // opacity: {
+    //   value: 0.6,
+    // },
+  },
+};
+
+const ORG_DASHBOARD_TEMPLATE_OVERLAY = {
+  config: {
+    legend: {
+      disable: true,
+    },
+    axis: {
+      // labels: false,
+      title: null,
+    },
+    style: {
+      cell: {
+        stroke: 'transparent',
+      },
+    },
+  },
+};
+
+const defaultFontStyleDict = {
+  titleFont: 'Source Sans Pro',
+  titleFontWeight: 'normal',
+  titleColor: globals.gray900,
+  labelFont: 'Source Sans Pro',
+  labelFontWeight: 'normal',
+  labelColor: globals.gray900,
+  labelSeparation: 5,
+};
+
+type ExprDimNameType = (typeof PlotState.EXPRESSION_DIM_NAMES)[number];
+
+function fixKeyForVegaTable(
+  key: string,
+  table: TableState.TableState,
+  opStore: OpStore
+) {
+  return PlotState.fixKeyForVega(
+    TableState.getTableColumnName(
+      table.columnNames,
+      table.columnSelectFunctions,
+      key,
+      opStore
+    )
+  );
+}
+
+function useIsOrgDashboard() {
+  return Object.keys(useContext(ActivityDashboardContext).frame).length > 0;
+}
+function useIsRepoInsightsDashboard() {
+  return Object.keys(useContext(RepoInsightsDashboardContext).frame).length > 0;
+}
```

Comment:
recommendation: a file called `hooks.ts` that has all the react hooks for this component. this is the standard organizational pattern for react hooks i've seen throughout the frontend

Reply:
My inclination is to only make a `hooks.ts` if the hook is used by multiple components or if it is large enough to benefit from being in a separate file - these ones here could almost be inlined, they just help with readability.